src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -38961,10 +39066,751 @@ static void F(object? x, object? y)
+            F(
+                t.Item2 = s,
+                v = t,
+                v.x.ToString(), // 2
+                v.y.ToString());
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (16,17): warning CS8602: Possible dereference of a null reference.
+                //                 u.y.ToString()) : // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "u.y").WithLocation(16, 17),
+                // (20,17): warning CS8602: Possible dereference of a null reference.
+                //                 v.x.ToString(), // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "v.x").WithLocation(20, 17));
+            comp.VerifyTypes();
+        }
+
+        [Fact]
+        [WorkItem(29970, "https://github.com/dotnet/roslyn/issues/29970")]
+        public void Tuple_Assignment_15()
+        {
+            var source =
+@"class C
+{
+    (object x, (string? z, object w) y) F;
+    static void M(C c, (object? x, (string z, object? w) y) t)
+    {
+        c.F = t; // 1
+        (object, (string?, object)) u = c.F/*T:(object! x, (string? z, object! w) y)*/;
+        c.F.x.ToString(); // 2
+        c.F.y.z.ToString();
+        c.F.y.w.ToString(); // 3
+        u.Item1.ToString(); // 4
+        u.Item2.Item1.ToString();
+        u.Item2.Item2.ToString(); // 5
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (6,15): warning CS8619: Nullability of reference types in value of type '(object? x, (string z, object? w) y)' doesn't match target type '(object x, (string? z, object w) y)'.
+                //         c.F = t; // 1
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "t").WithArguments("(object? x, (string z, object? w) y)", "(object x, (string? z, object w) y)").WithLocation(6, 15),
+                // (8,9): warning CS8602: Possible dereference of a null reference.
+                //         c.F.x.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c.F.x").WithLocation(8, 9),
+                // (10,9): warning CS8602: Possible dereference of a null reference.
+                //         c.F.y.w.ToString(); // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c.F.y.w").WithLocation(10, 9),
+                // (11,9): warning CS8602: Possible dereference of a null reference.
+                //         u.Item1.ToString(); // 4
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "u.Item1").WithLocation(11, 9),
+                // (13,9): warning CS8602: Possible dereference of a null reference.
+                //         u.Item2.Item2.ToString(); // 5
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "u.Item2.Item2").WithLocation(13, 9));
+            comp.VerifyTypes();
+        }
+
+        [Fact]
```

Comment:
>[Fact] [](start = 8, length = 6)

If we don't have one already, maybe test a nullable tuple too? `(string, string?)? t = (null, ""); ...`